src/sage/combinat/partition.py

```diff
@@ -7596,8 +7596,19 @@ def __classcall_private__(cls, n, parts):
             sage: P2 = Partitions(4, parts_in=(1,2))
             sage: P is P2
             True
+
+        Ensure that :issue:`38640` is fixed::
+
+            sage: list(Partitions(4,parts_in=vector(QQ,[2,4])))
+            [[4], [2, 2]]
+            sage: list(Partitions(4,parts_in=vector(QQ,[2,1/4])))
+            Traceback (most recent call last):
+            ...
+            TypeError: no conversion of this rational to integer
+            sage: list(Partitions(4,parts_in=vector(ZZ,[2,4])))
+            [[4], [2, 2]]
         """
-        parts = tuple(sorted(parts))
+        parts = tuple(sorted(map(ZZ,parts)))
```

Comment:
```suggestion
        parts = tuple(sorted(set(map(ZZ,parts))))
```
Let's also be helpful here in case there are repeats.

Reply:
Thank you for the comment! The change has been applied. I will now merge the branch with the newest beta and wait for a positive review.